rollup.config.js

```diff
@@ -11,4 +13,12 @@ export default {
 			banner: '// @ts-self-types="./index.d.ts"',
 		},
 	],
+	plugins: [
+		copy({
+			targets: [
+				{ src: "src/types.ts", dest: "dist/cjs" },
+				{ src: "src/types.ts", dest: "dist/esm" },
```

Comment:
I now see `dist/(cjs|esm)/types.d.ts` and `dst/(cjs|esm)/types.ts` files locally. They're identical other than comments and an auto-generated `export {}`. I don't see anything importing from the `types.d.ts`. Is the duplication intentional?

FWIW I believe it's more common to have just `.d.ts` files. My instinct is that the expected path here would be to just have `dist/(cjs|esm)/types.d.ts` and typedefs use `{import("./types")}`.

Reply:
Because we run `tsc` on `dist/esm` and `dist/cjs`, `types.ts` needs to be present in both of those directories in order for the project to compile.

Reply:
Gotcha. I tried renaming `src/types.ts` to `src/types.d.ts` with a find-and-replace. I think it's all working here: https://github.com/JoshuaKGoldberg/json/commit/631810d666a5bd376955a0e35695398b83c3fd4f. At least `npm run build` passes and the imports all work in editor.